pkg/ec2helper/ec2helper.go

```diff
@@ -975,15 +1008,30 @@ func getRunInstanceInput(simpleConfig *config.SimpleInfo, detailedConfig *config
 			}
 		}
 
-		// Set auto-termination timer
-		if IsLinux(*detailedConfig.Image.PlatformDetails) && simpleConfig.AutoTerminationTimerMinutes > 0 {
+		// Set auto-termination
+		setAutoTermination = IsLinux(*detailedConfig.Image.PlatformDetails) && simpleConfig.AutoTerminationTimerMinutes > 0
+		if setAutoTermination {
 			input.InstanceInitiatedShutdownBehavior = aws.String("terminate")
-			userData := []byte(fmt.Sprintf("#!/bin/bash\necho \"sudo poweroff\" | at now + %d minutes",
-				simpleConfig.AutoTerminationTimerMinutes))
-			input.UserData = aws.String(base64.StdEncoding.EncodeToString(userData))
 		}
 	}
 
+	if simpleConfig.UserDataFilePath != "" {
```

Comment:
looks like we need an else block here to continue to support `autoTerminationTimer` without userData?